andremleblanc/tictactoe
# Implement Tictactoe

**Feature**

Add an unbeatable Tic Tac Toe implementation leveraging the existing framework.

**Description**

This is a partially complete implementation. The UI of the app can be accessed on the root of the frontend `localhost:3000`, the game can be played, and the computer can't be beaten (and may heckle you a little). I've left comments in parts of the code where I would love some feedback and I've left a list of remaining work to make this PR feature complete. 

**Testing Instructions**

1. Start the app [following the instructions in the readme](https://github.com/andremleblanc/tictactoe#getting-started).
2. Go to [http://localhost:3000/](http://localhost:3000/).
3. Play the game and ensure you can't beat the computer.

**Feature List**

- [x] Tic Tac Toe UI: Allow players to play tic tac toe through the frontend.
- [x] Unbeatable AI: Have players compete against an AI that cannot be beaten.
- [ ] Difficulty: Allow players to chose a difficulty for the AI.
- [ ] First Player Settings: Allow players to select who goes first (themselves or the AI).
- [ ] Mark the Winning Line: Show which row was the winning row.
- [ ] Tests: Ensure there is adequate test coverage.
- [ ] Edge Cases: Ensure edge cases are covered (I've covered some, like disabling squares that have been clicked, but I'd imagine there are more to cover).
